archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java

```diff
@@ -509,4 +509,18 @@ public void check(JavaClass implementationClass, ConditionEvents events) {
                     .should(accessTargetWhere(JavaAccess.Predicates.target(annotatedWith(Deprecated.class))).as("access @Deprecated members"))
                     .orShould(dependOnClassesThat(annotatedWith(Deprecated.class)).as("depend on @Deprecated classes"))
                     .because("there should be a better alternative");
+
+    /**
+     * A rule checking that no classes uses old date and time classes and point out to use the Java 8 time API.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchRule OLD_DATE_AND_TIME_CLASSES_SHOULD_NOT_BE_USED =
+            noClasses()
+                    .should().dependOnClassesThat().haveFullyQualifiedName("java.sql.Date")
+                    .orShould().dependOnClassesThat().haveFullyQualifiedName("java.sql.Time")
+                    .orShould().dependOnClassesThat().haveFullyQualifiedName("java.sql.Timestamp")
+                    .orShould().dependOnClassesThat().haveFullyQualifiedName("java.util.Calendar")
+                    .orShould().dependOnClassesThat().haveFullyQualifiedName("java.util.Date")
+                    .as("since Java 8 (and JavaEE 7 if JPA is needed) java.time-API should be used.")
+                    .because("since Java 8 (and JavaEE 7 if JPA is needed) java.time-API should be used.");
```

Comment:
If you use the same text in `as` and `because`, you essentially duplicate it in the final rule description:

>  Rule '`since Java 8 (and JavaEE 7 if JPA is needed) java.time-API should be used., because since Java 8 (and JavaEE 7 if JPA is needed) java.time-API should be used.'` was violated [...]

How about

```suggestion
                    .as("java.time API should be used")
                    .because("legacy date/time APIs have been replaced since Java 8  (JSR 310)");
```

which gives the final rule description

> Rule '`java.time API should be used, because legacy date/time APIs have been replaced since Java 8`' was violated [...]

* without full stop before `, because`,
* without the reference to Java EE 7 & JPA, because it might be confusing if the general coding rule is used in an context unrelated to Java EE/JPA, but
* with a reference to  JSR 310, which might help unexperienced developers to find more information.

Reply:
For your convenience, I've prepared this suggestion on the [`dateutil-review` branch](https://github.com/TNG/ArchUnit/commits/dateutil-review/).

Reply:
Thank you for your suggestion. I'm fine with that. In my initial version I added the JPA information, because we have that at work, that Java 8 is available, but not JavaEE 7.
Also run spotless tasks, so that should fit too now.